runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java

```diff
@@ -255,11 +262,42 @@ public void onGameTick(final GameTick event)
 
 		if (clue instanceof LocationsClueScroll)
 		{
-			final List<WorldPoint> locations = ((LocationsClueScroll) clue).getLocations();
+			final WorldPoint[] locations = ((LocationsClueScroll) clue).getLocations();
 
-			if (!locations.isEmpty())
+			if (locations.length > 0)
 			{
-				addMapPoints(locations.toArray(new WorldPoint[locations.size()]));
+				addMapPoints(locations);
+			}
+
+			if (clue instanceof ObjectClueScroll)
+			{
+				List<Integer> objectIds = ((ObjectClueScroll) clue).getObjectIds();
+
+				if (!objectIds.isEmpty()
+					&& locations.length > 0)
+				{
+					for (WorldPoint location : locations)
+					{
+						final LocalPoint localLocation = LocalPoint.fromWorld(client, location);
+
+						if (localLocation != null)
+						{
+							final Scene scene = client.getScene();
+							final Tile[][][] tiles = scene.getTiles();
+							final Tile tile = tiles[client.getPlane()][localLocation.getSceneX()][localLocation.getSceneY()];
+
+							objectsToMark = Arrays.stream(tile.getGameObjects())
+								.filter(object -> object != null && objectIds.contains(object.getId()))
```

Comment:
Make getObjectIds return an array and use ArrayUtils.contains instead of List#contains here.

Reply:
I believe there's a reason I couldn't easily do that, but will look into this again and report back once I do.